runtime/src/app/app.ts

```diff
@@ -185,11 +194,17 @@ export async function navigate(target: Target, id: number, noscroll?: boolean, h
 	prefetching = null;
 
 	const token = current_token = {};
-	const { redirect, props, branch } = await loaded;
+	const loaded_result = await loaded;
+	const { redirect } = loaded_result;
 	if (token !== current_token) return; // a secondary navigation happened while we were loading
 
-	await render(redirect, branch, props, target.page);
-	if (document.activeElement) document.activeElement.blur();
+	if (redirect) {
+		await goto(redirect.location, { replaceState: true });
+	} else {
+		const { props, branch } = loaded_result;
+		await render(branch, props, target.page);
+	}
+	if (document.activeElement && (document.activeElement instanceof HTMLElement)) document.activeElement.blur();
```

Comment:
Might make sense to highlight that this is a fix in the commit message

Reply:
this is pretty minor. I'll end up squashing the commits when I merge, so I don't think there's a need. if this were bigger I'd send it as a separate PR

Reply:
Ok, I don't know precedent for this project. I think helping people comb through commits that might fix their bugs in prod can make them more eager to update ;)

Just a nit on my part